rust/src/mqtt/mqtt.rs

```diff
@@ -553,8 +553,8 @@ impl MQTTState {
                     let _pdu = Frame::new(
                         flow,
                         &stream_slice,
-                        input,
-                        input.len() as i64,
+                        current,
+                        current.len() as i64,
```

Comment:
This should not be `current.len()` as current may hold multiple PDUs, but rather `start.len() - rem.len()`